lib/map/security_token/service.rb

```diff
@@ -57,11 +57,17 @@ def parse_and_raise_error(e, icn, application)
 
       def parse_response(response, application, icn)
         response_body = response.body
+        if response_body['expires_in'].to_i > config.max_token_duration
+          raise Errors::InvalidTokenDurationError,
+                "#{config.logging_prefix} token failed, token duration exceeds maximum"
+        end
```

Comment:
should we move this check to a separate method? I think it would make it a bit easier to read